### PR TITLE
ceph: use "github.com/pkg/errors"

### LINKED ISSUE #4054: Start using errors.Wrap() instead of fmt.Errorf
Rook could benefit from using `errors.Wrap()` as well as `errors.Wrapf()` so the error formatting is handled by the package. Many projects under the Kubernetes umbrella use it. Also:

* from a formatting standpoint, the code will be cleaner
* reviews will be more focus since reviewers won't have to explicitly explain on to handle errors and which interface type to use (`%v` versus `+%` for example)

Currently, they are around 1414 occurrence of `fmt.Errorf` which should be changed so we need to decide as a group :).
This means using `"github.com/pkg/errors"`.

### PR DESCRIPTION
**Pull request by leseb**
We now use the error package.
Kubernetes errors have been renamed kerrors since they are lower than
'errors'.

Closes: https://github.com/rook/rook/issues/4054
Signed-off-by: Sébastien Han <seb@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #4054

**Checklist:**

- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](https://github.com/rook/rook/blob/master/CONTRIBUTING.md#comments)
- [ ] Add the flag for skipping the CI if this PR does not require a build. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for more details.

[test ceph]

### CODE REVIEW

**Comment by BlaineEXE** [comment]
Can you explain a little more about what benefits Rook gets from using the new errors package instead of builtins?

**Comment by leseb** [comment]
> Can you explain a little more about what benefits Rook gets from using the new errors package instead of builtins?
> 
> Especially given this part of the Readme, `github.com/pkg/errors` seems like it is not a good change to make. https://github.com/pkg/errors#roadmap
> 
> > With the upcoming Go2 error proposals this package is moving into maintenance mode.

Isn't https://github.com/rook/rook/issues/4054 enough?
Some highlights:

* code consistency (cosmetic)
* error formatting will be standardized thanks to `errrors.Wrapf()`

Also from a reviewer standpoint, this will hopefully something we won't have to review anymore, the `%v` versus `%+v`, the position of `err` in the error message etc...

File: pkg/daemon/ceph/client/erasure-code-profile.go [hunk 1]
```diff
@@ -105,7 +106,7 @@ func DeleteErasureCodeProfile(context *clusterd.Context, clusterName string, era
 	cmd.JsonOutput = false
 	buf, err := cmd.Run()
 	if err != nil {
-		return fmt.Errorf("failed to delete erasure-code-profile %s. Output: %s. Error: %+v", erasureCodeProfile, string(buf), err)
```
Comment (BlaineEXE) [inline pkg/daemon/ceph/client/erasure-code-profile.go]: why remove `string()` from `string(buf)`? The string prints more nicely.

File: pkg/daemon/ceph/client/image.go [hunk 2]
```diff
@@ -139,7 +140,7 @@ func ExpandImage(context *clusterd.Context, clusterName, name, poolName, monitor
 	}
 	output, err := ExecuteRBDCommandWithTimeout(context, clusterName, args)
 	if err != nil {
-		return fmt.Errorf("failed to resize image %s in pool %s, output: %s, err: %+v", name, poolName, string(output), err)
+		return errors.Wrapf(err, "failed to resize image %s in pool %s, output: %s", name, poolName, output)
```
Comment (BlaineEXE) [inline pkg/daemon/ceph/client/image.go]: `string()` is lost from `string(output)` here

**Comment by leseb** [comment]
@BlaineEXE @travisn as we are approaching 1.2 I've just rebased this.

**Pull request merged**